lib/realtime/repo.ex

```diff
@@ -19,11 +19,31 @@ defmodule Realtime.Repo do
   @doc """
   Converts a Postgrex.Result into a given struct
   """
-  @spec pg_result_to_struct(Postgrex.Result.t(), module()) :: [struct()]
-  def pg_result_to_struct(%Postgrex.Result{rows: rows, columns: columns} = res, struct) do
-    Enum.map(rows, &Realtime.Repo.load(struct, Enum.zip(columns, &1)))
+  @spec result_to_single_struct({:ok, Postgrex.Result.t()} | {:error, any()}, module()) ::
+          {:ok, struct()} | {:ok, nil} | {:error, any()}
+  def result_to_single_struct({:ok, %Postgrex.Result{rows: [row], columns: columns}}, struct) do
+    {:ok, Realtime.Repo.load(struct, Enum.zip(columns, row))}
   end
 
+  def result_to_single_struct({:ok, %Postgrex.Result{rows: []}}, _),
+    do: {:ok, nil}
+
+  def result_to_single_struct({:ok, %Postgrex.Result{rows: rows}}, _),
+    do: raise("expected at most one result but got #{length(rows)} in result")
```

Comment:
double checking if it's okay to raise an error here or it would be better to return a response like `{:error, msg}`